python/cudf/cudf/core/column/string.py

```diff
@@ -682,28 +682,20 @@ def serialize(self):
     @classmethod
     def deserialize(cls, header, frames):
         # Deserialize the mask, value, and offset frames
-        arrays = []
-
-        for each_frame in frames:
-            if hasattr(each_frame, "__cuda_array_interface__"):
-                each_frame = cuda.as_cuda_array(each_frame)
-            elif isinstance(each_frame, memoryview):
-                each_frame = np.asarray(each_frame)
-                each_frame = cudautils.to_device(each_frame)
-
-            arrays.append(libcudf.cudf.get_ctype_ptr(each_frame))
+        buffers = [Buffer(each_frame) for each_frame in frames]
+        ptrs = [ctypes.c_uint64(buf.ptr) for buf in buffers]
 
         if header["null_count"] > 0:
-            nbuf = arrays[2]
+            nbuf = ptrs[2]
         else:
             nbuf = None
 
         # Use from_offsets to get nvstring data.
         # Note: array items = [nbuf, sbuf, obuf]
         scount = header["nvstrings"]
         data = nvstrings.from_offsets(
-            arrays[0],
-            arrays[1],
+            ptrs[0],
+            ptrs[1],
             scount,
             nbuf=nbuf,
```

Comment:
Seem to be getting a segfault in this function. Any ideas what we might be missing?

https://gpuci.gpuopenanalytics.com/blue/rest/organizations/jenkins/pipelines/rapidsai/pipelines/gpuci/pipelines/cudf/pipelines/prb/pipelines/cudf-gpu-build/runs/12711/log/?start=0

Reply:
NVstrings expects normal Python integers and not ctypes. I'm guessing it's just not handled properly at the C++ where it uses the Python C API.

Reply:
I changed it to `int` just to see if that worked.

Reply:
Ha! That worked 😄